### PR TITLE
feat: add data source and oracle script messages support

### PR DESCRIPTION
**Pull request by prin-r**
## Please ensure the following requirements are met before submitting a pull request:

- [x] The pull request is targeted against the correct target branch
- [ ] The pull request is linked to an issue with appropriate discussion and an accepted design OR is linked to a spec that describes the work.
- [ ] The pull request includes a description of the implementation/work done in detail.
- [x] The pull request includes any and all appropriate unit/integration tests
- [ ] You have added a relevant changelog entry to `CHANGELOG_UNRELEASED.md`
- [ ] You have re-reviewed the files affected by the pull request (e.g. using the `Files changed` tab in the Github PR explorer)

Please see this reference implementation 
https://github.com/bandprotocol/bandchain/pull/3231

### CODE REVIEW

**Comment by evilpeach** [comment]
@prin-r Please check test failed on node 12

**Review by evilpeach — CHANGES_REQUESTED** [review]
Let's divide messages( in message.test.ts ) into its own.
Like `__tests_/messages/send.test.ts`

File: __tests__/mockDsOs/example_data_source.py [hunk 1]
```diff
@@ -0,0 +1,73 @@
+#!/usr/bin/env python3
```
Comment (evilpeach) [inline __tests__/mockDsOs/example_data_source.py:1]: change folder `mockDsOs` to `mock_ds_os`

File: package.json [hunk 2]
```diff
@@ -45,6 +45,7 @@
     "bip39": "^3.0.2",
     "bitcoinjs-lib": "^5.2.0",
     "crypto": "^1.0.1",
+    "fs": "^0.0.1-security",
```
Comment (evilpeach) [inline package.json:48]: Why we have to added this?

File: src/message.ts [hunk 3]
```diff
@@ -142,3 +143,269 @@ export class MsgDelegate extends Msg {
...
+      },
+    }
+  }
+
+  getSender() {
+    return this.sender
+  }
+
+  validate() {
+    if (this.name.length <= 0) throw Error('Invalid data source name')
```
Comment (evilpeach) [inline src/message.ts:196]: Let's create custom error for this like valuator 
[Ref](https://github.com/bandprotocol/bandchain.js/pull/7)

**Review by evilpeach — CHANGES_REQUESTED** [review]

File: src/message.ts (hunk 3 above)
Comment (prin-r) [reply src/message.ts]: done

File: package.json (hunk 2 above)
Comment (prin-r) [reply package.json]: done

File: __tests__/mockDsOs/example_data_source.py (hunk 1 above)
Comment (prin-r) [reply __tests__/mockDsOs/example_data_source.py]: done

File: src/message.ts [hunk 4]
```diff
@@ -150,3 +157,272 @@ export class MsgDelegate extends Msg {
...
+    this.dataSourceID = dataSourceID
+    this.name = name
+    this.description = description
+    this.script = script
+  }
+
+  _replaceEmpty(placeholder: string, s?: string) {
+    if (s === undefined) return placeholder
+    return s
+  }
```
Comment (evilpeach) [inline src/message.ts:249]: Let's move this function to `helper.ts`

File: src/message.ts [hunk 5]
```diff
@@ -150,3 +157,272 @@ export class MsgDelegate extends Msg {
...
+    this.script = script
+    this.description = description
+    this.schema = schema
+    this.sourceCodeURL = sourceCodeURL
+  }
+
+  _replaceEmpty(placeholder: string, s?: string) {
+    if (s === undefined) return placeholder
+    return s
+  }
```
Comment (evilpeach) [inline src/message.ts:317]: Let's move this function to `helper.ts`

File: src/message.ts [hunk 6]
```diff
@@ -150,3 +157,272 @@ export class MsgDelegate extends Msg {
...
+    return Buffer.from(fs.readFileSync(this.script, 'utf8')).toString('base64')
+  }
+
+  asJson() {
+    return {
+      type: 'oracle/EditOracleScript',
+      value: {
+        oracle_script_id: this.oracleScriptID,
+        owner: this.owner.toAccBech32(),
+        name: this._replaceEmpty('[do-not-modify]', this.name),
```
Comment (evilpeach) [inline src/message.ts:401]: Let's declare `[do-not-modify]` as a variable on `constant.ts`

**Review by evilpeach — CHANGES_REQUESTED** [review]

**Review by evilpeach — CHANGES_REQUESTED** [review]
You should test functions on React app as well.

**Comment by evilpeach** [comment]
Move to #25 #26

**Pull request closed without merging**